code/datums/components/lightdrawn.dm

```diff
@@ -0,0 +1,135 @@
+
+/**
+  * Drawn to light component
+  *
+  * This component adds game mechanics to emulate the moth lamp meme
+  */
+/datum/component/lightdrawn
+	var/next_move /// Next time we can move
+	var/move_delay = 2 SECONDS /// How often we move
+	var/next_message /// Limiter to stop LAMP spam
+	var/light_mood = 100 /// current light mood
+	var/light_mood_threshold = 50 /// threshold below which lamp memes happen
+	var/light_level_threshold = 1 /// threshold to compare with [/atom/movable/lighting_object/var/cached_max]
+	var/mesmerised = FALSE /// boolean to track druggy/mesmerised overlay add/remove
+
+/datum/component/lightdrawn/Initialize()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	. = ..()
+
+	START_PROCESSING(SSprocessing, src)
+
+/datum/component/lightdrawn/Destroy(force, silent)
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/datum/component/lightdrawn/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_LIVING_FLASH_ACT, VARSET_CALLBACK(src, light_mood, 100))
+
+/datum/component/lightdrawn/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, COMSIG_LIVING_FLASH_ACT)
+
+/datum/component/lightdrawn/process()
+	var/turf/T = get_turf(parent)
+	if(!T)
+		return
+
+	var/mob/living/living_parent = parent
+
+	var/tint_factor = 1
+	var/brightness_mult = 1
+
+	if(iscarbon(parent))
+		var/mob/living/carbon/C = parent
+		var/totaltint = 0
+		if(C?.glasses)
+			totaltint += C.glasses.tint
+			brightness_mult = min(1, (2 - C.glasses.darkness_view) * 0.33)
+		var/obj/item/organ/eyes/eye = C.getorganslot(ORGAN_SLOT_EYES)
+		if(eye?.tint)
+			totaltint += eye.tint
+		tint_factor = 1/(1 + totaltint)
+
+	var/apparent_brightness = T.lighting_object.cached_max * tint_factor * brightness_mult
+
+	// update their mood
+	if(living_parent.loc != T) // inside something
+		apparent_brightness = 0 // its dark inside things

```

Comment:
wouldn't it be bright inside of a closet if you have a light in your pocket/hand or something?

Reply:
its too complicated to try to deal with that tbh

Reply:
iterating through the entire recursive contents of a container to look for light sources is dumb given the point of this component is ambient light levels on the turf

Reply:
then why not just use the ambient light levels of the locker's turf, instead of adding a snowflake case for if you're in a locker, clown car, void hole, etc.?

Reply:
its not really snowflake if its generic 🤔  it would be snowflake if he did istype